central_web_server/handle_database.js

```diff
@@ -104,8 +104,8 @@ async function addSurveyResults(surveyData) {
     .toString()
   let surveyResult = await connection.query(surveyQuery);
   surveyId = surveyResult.insertId;
-  
   let surveyResults = [];
+  if (!surveyData.found_species.length) return true;
```

Comment:
This will not close the connection, only run the code below if the found_species.length is >0 and use the connection.end() at the end of the function.